src/interpreter.ts

```diff
@@ -0,0 +1,176 @@
+import { Path, PathType } from './path';
+import { Code, GCodeCommand } from './gcode-parser';
+import { Machine } from './machine';
+
+export class Interpreter {
+  execute(commands: GCodeCommand[], machine = new Machine()): Machine {
+    commands.forEach((command) => {
+      if (command.code !== undefined) {
+        this[command.code](command, machine);
+      }
+    });
+
+    return machine;
+  }
+
+  G0(command: GCodeCommand, machine: Machine): void {
+    const { x, y, z, e } = command.params;
+    const { state } = machine;
+
+    let lastPath = machine.paths[machine.paths.length - 1];
+    const pathType = e ? PathType.Extrusion : PathType.Travel;
+
+    if (lastPath === undefined || lastPath.travelType !== pathType) {
+      lastPath = this.breakPath(machine, pathType);
+    }
+
+    state.x = x || state.x;
+    state.y = y || state.y;
+    state.z = z || state.z;
+
+    lastPath.addPoint(state.x, state.y, state.z);
+  }
+
+  G1 = this.G0;
```

Comment:
Aliases!

Reply:
aha! nice!
this can even be moved up, before the method declaration, so we can have all aliases together and visible